plugin_tests/source_mapnik_test.py

```diff
@@ -285,4 +285,111 @@ def testPixel(self):
             })
         self.assertStatusOk(resp)
         self.assertEqual(resp.json, {
-            'r': 225, 'g': 100, 'b': 98, 'a': 255, 'bands': {'1': 74.0, '2': 80.0, '3': 84.0}})
+            'r': 225, 'g': 100, 'b': 98, 'a': 255, 'bands': {'1': 77.0, '2': 82.0, '3': 84.0}})
+        # Test with projection units
+        resp = self.request(
+            path='/item/%s/tiles/pixel' % itemId, user=self.admin,
+            params={
+                'left': -13132910,
+                'top': 4010586,
+                'projection': 'EPSG:3857',
+                'units': 'EPSG:3857',
+            })
+        self.assertStatusOk(resp)
+        self.assertEqual(resp.json, {
+            'r': 77, 'g': 82, 'b': 84, 'a': 255, 'bands': {'1': 77.0, '2': 82.0, '3': 84.0}})
+        resp = self.request(
+            path='/item/%s/tiles/pixel' % itemId, user=self.admin,
+            params={
+                'left': -117.975,
+                'top': 33.865,
+                'projection': 'EPSG:3857',
+                'units': 'WGS84',
+            })
+        self.assertStatusOk(resp)
+        self.assertEqual(resp.json, {
+            'r': 77, 'g': 82, 'b': 84, 'a': 255, 'bands': {'1': 77.0, '2': 82.0, '3': 84.0}})
+        # When the tile has a different projection, the pixel is the same as
+        # the band values.
+        resp = self.request(
+            path='/item/%s/tiles/pixel' % itemId, user=self.admin,
+            params={
+                'left': -13132910,
+                'top': 4010586,
+                'units': 'EPSG:3857',
+            })
+        self.assertStatusOk(resp)
+        self.assertEqual(resp.json, {
+            'r': 77, 'g': 82, 'b': 84, 'a': 255, 'bands': {'1': 77.0, '2': 82.0, '3': 84.0}})
+
+    def testSourceErrors(self):
+        from girder.plugins.large_image.tilesource import TileSourceException
+        from girder.plugins.large_image.tilesource.mapniksource import MapnikTileSource
+        filepath = os.path.join(
+            os.path.dirname(__file__), 'test_files', 'rgb_geotiff.tiff')
+        with six.assertRaisesRegex(self, TileSourceException, 'must not be geographic'):
```

Comment:
Why is this still getting raised?

Reply:
Tiles have to be in a non-geographic projection.  Units can be in any projection, but the `projection` for a tile can't be, since it needs to map to a 2D conformal space.

Reply:
Oh ok, now I got it.